src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -3534,14 +3537,81 @@ private static bool DirectoryInfoHasChildItems(DirectoryInfo directory)
                 }
                 else // Copy-Item local
                 {
+                    if (Context != null)
+                    {
+                        if (Context.ExecutionContext.SessionState.PSVariable.Get(SpecialVariables.ProgressPreferenceVarPath.ToString()).Value is ActionPreference progressPreference)
+                        {
+                            if (progressPreference == ActionPreference.Continue)
+                            {
+                                // we use a thread to gather the total files so that copying can start right away
+                                Thread t = new Thread(() =>
+                                {
+                                    GetTotalFiles(path, recurse);
+                                });
+                                t.Start();
+                                _copyStartTime = DateTime.Now;
```

Comment:
`DateTime.Now` is very expensive (since use conversion to local time) and really we have no need to have absolute time, we need relative time. We could use static `StopWatch.GetTimestamp()` here and `public static [TimeSpan] GetElapsedTime(long startingTimestamp)` below.

Since we have start time we can implement easily startup delay I asked previously - not show progress bar first 5-10-15 seconds. (Although ideally such a thing would have to be implemented in the ProgressBar API itself in generic way).

And if we say about async/await we can easily move coping to another thread and address freezing screen issue for coping large files.

Reply:
I think having copy on another thread or even using multiple threads is out of scope of this PR